Branch-SDK/src/io/branch/referral/ServerRequestCreateUrl.java

```diff
@@ -62,6 +63,11 @@ public ServerRequestCreateUrl(Context context, final String alias, final int typ
 
         linkPost_ = new BranchLinkData();
         try {
+            if (canonicalID != null && canonicalID.length() > 0) {
+                JSONObject paramsJson = new JSONObject(params);
```

Comment:
Aren't you assembling the params in the Universal Object? Why do you need to insert this here?